### PR TITLE
MLDB-1827 fix NULL propagation for base math operators

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: sql/sql_expression_operations.cc [hunk 1]
```diff
@@ -1171,8 +1171,12 @@ static CellValue binaryModulus(const CellValue & la, const CellValue & ra)
 struct BinaryPlusOp {
     static CellValue apply(const CellValue & l, const CellValue & r)
     {
-        if (l.empty() || r.empty())
+        if (l.empty()) {
             return l;
+        }
+        if (r.empty()) {
+            return r;
+        }
```
Comment (FinchPowers) [inline sql/sql_expression_operations.cc]: @jeremybarnes Any timestamp logic to take into account here? Should it return the max ts between l and r?

Comment (jeremybarnes) [inline sql/sql_expression_operations.cc]: Yes

Comment (jeremybarnes) [inline sql/sql_expression_operations.cc]: What all of these should be is:

```
if (l.empty() || r.empty()) {
    return ExpressionValue::null(calcTs(l, r));
}
```

Comment (FinchPowers) [inline sql/sql_expression_operations.cc]: I updated the test and there was nothing to do after all. Ts are not part of cell values.

Comment (jeremybarnes) [inline sql/sql_expression_operations.cc]: In that case, the logic is more convoluted than necessary.  We can just do if (l.empty() or r.empty()) return CellValue();

**Comment by jeremybarnes** [comment]
Consider simplifying the null logic.  +1 after that.

**Pull request merged**